inventory-api/src/main/scala/com/normation/inventory/services/provisionning/CheckInventoryDigest.scala

```diff
@@ -150,18 +150,42 @@ class InventoryDigestServiceV1(
    * or if there was no inventory before, it will look for the key in the received inventory
    */
   def getKey (receivedInventory  : InventoryReport) : Box[(PublicKey, KeyStatus)] = {
-    for {
-      nodeInventory <- repo.get(receivedInventory.node.main.id) match {
-        case Full(storedInventory) =>
-          Full(storedInventory.node)
-        case _ =>
-          Full(receivedInventory.node)
+
+    def treatKey (status: KeyStatus, node : NodeInventory) : Box[(PublicKey, KeyStatus)]= {
```

Comment:
You don't use keystatus at all in that method, so you are introducing a non needed coupling here. 
Could you remove it, and name the method "extractKey" ? And use a extractKey.map( k => (k, status) ) at the end ?